src/napari_imagej/utilities/_module_utils.py

```diff
@@ -302,15 +295,17 @@ def _pure_module_outputs(
     for output_entry in outputs.entrySet():
         name = str(output_entry.getKey())
         # Ignore preallocated outputs if they were provided by the user
-        if module.getInfo().getInput(name) in user_inputs and module.getInput(name):
-            continue
         output = ij().py.from_java(output_entry.getValue())
         # Add arraylike outputs as images
-        if ij().py._is_arraylike(output):
+        if is_arraylike(output):
+            if module.getInfo().getInput(name) in user_inputs and module.getInput(name):
+                continue
             layer = Layer.create(data=output, meta={"name": name}, layer_type="image")
             layer_outputs.append(layer)
         # Add Layers directly
         elif isinstance(output, Layer):
+            if module.getInfo().getInput(name) in user_inputs and module.getInput(name):
+                continue
```

Comment:
What's going on here?

I remember that the effect of the change is that mutable `RealType`s are treated as pure outputs. But could we do this in a cleaner way?

Reply:
It's cleaned up!